### PR TITLE
fix(423): Add job descriptions to validator

### PR DESCRIPTION
**Pull request by joelseq**
- Add SCHEMA_DESCRIPTION to SCHEMA_JOB_PERMUTATION and SCHEMA_OUTPUT
- Modify validator.input.yaml and validator.output.yaml to make sure
  description and annotations work (related to annotations PR https://github.com/screwdriver-cd/data-schema/pull/145).

Part of #149

### CODE REVIEW

File: test/data/validator.output.yaml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 jobs:
   main:
+  - annotations:
```
Comment (d2lam) [inline test/data/validator.output.yaml]: I'm kinda confused. Shouldn't this be in the same level as `settings` and `environments`?

File: api/validator.js [hunk 2]
```diff
@@ -44,7 +44,8 @@ const SCHEMA_JOBS = Joi.object()
 
 const SCHEMA_OUTPUT = Joi.object()
     .keys({
-        annotations: Annotations.annotations,
+        annotations: Job.annotations,
+        description: Job.description,
```
Comment (minz1027) [inline api/validator.js]: Why we need it here? Should be inside the job permutation already right?

File: test/data/validator.input.yaml [hunk 3]
```diff
@@ -1,6 +1,9 @@
 yaml: |
   jobs:
       main:
+          annotations:
+              beta.screwdriver.cd/auto_pr_builds: fork-only
+          description: "This is a description"
```
Comment (d2lam) [inline test/data/validator.input.yaml]: I'm kinda confused. Shouldn't this be in the same level as settings and environments?

File: test/data/validator.output.yaml [hunk 4]
```diff
@@ -1,5 +1,8 @@
...
+      beta.screwdriver.cd/auto_pr_builds: fork-only
+  - description: "This is a description"
```
Comment (d2lam) [inline test/data/validator.output.yaml]: I'm kinda confused. Shouldn't this be in the same level as settings and environments?

**Pull request merged**